### ISSUE TITLE
Make allowedPages configurable

### ISSUE THREAD

**Issue by peter-** [filesender/filesender#1261]
Unless there's an existing method that would enable this I think it would make sense to make `allowedPages()` in `classes/utils/GUI.class.php` somewhat configurable:

I recently noticed that another FileSender instance displayed `/?s=privacy` in the menu bar even when not yet authenticated -- which makes sense to me: to show information related to privacy / data protection even before someone logs in to the system. (Maybe the terms are such that someone decides to actually forgo logging in. Hopefully that's not ever the case, of course. Either way, enabling people to know the terms beforehand only seems proper.)

### Generic

I guess we could add the following to `includes/ConfigDefaults.php`

```php
'allowed_pages' => array('download', 'translate_email', 'logout', 'exception', 'help', 'about'),
```
and change `GUI.class.php` to make use of it:

```diff
--- a/classes/utils/GUI.class.php
+++ b/classes/utils/GUI.class.php
@@ -388,7 +388,7 @@ class GUI
             }
             
             // Always accessible pages
-            foreach (array('download', 'translate_email', 'logout', 'exception', GUIPages::HELP, GUIPages::ABOUT) as $p) {
+            foreach (Config::get('allowed_pages') as $p) {
                 self::$allowed_pages[] = $p;
             }
```
Then deployers could override the `$allowed_pages` array in their `config.php` as needed.  
Setting `$config["maintenance"] = true` still does the expected thing and only allows the 'maintenance' page (even though the $allowed_pages array functionally now overlaps with the specific 'maintenance' setting.)

(Note that in other places, e.g. `user_pages`, a different method is being used to de-/select individual pages.)

### Specific

Note that I currently can't imagine making any *other* pages publicly accessible, so a more specific variant of this request would be to only make the inclusion of `/?s=privacy` dynamic:

```diff
--- a/classes/utils/GUI.class.php
+++ b/classes/utils/GUI.class.php
@@ -391,6 +391,10 @@ class GUI
             foreach (array('download', 'translate_email', 'logout', 'exception', GUIPages::HELP, GUIPages::ABOUT) as $p) {
                 self::$allowed_pages[] = $p;
             }
+
+            if (Config::get('allow_pages_privacy')) {
+                self::$allowed_pages[] = 'privacy';
+            }
             
             if (Config::get('maintenance')) {
                 self::$allowed_pages = array('maintenance');
```

### Related note wrt page constants

Looking at `classes/utils/GUI.class.php` I wonder why only some of the pages are being referred to by their constant (as defined in `classes/constants/GUIPages.class.php`, e.g. `GUIPages::HELP` and `GUIPages::ABOUT` above) while others are directly referenced as strings (e.g. `download`, `translate_email`, etc.) even though constants for all of those exist.

To put this another way: The only use of the constants in `GUIPages.class.php` I can find is within `GUI.class.php` and even there it's only for less than half of the defined pages. So maybe use of the constants should be phased out -- or alternatively used consistently whenever pages are being references? (Then the above diff would have to be changed to add `GUIPages::PRIVACY` to the array instead.)

**Comment by monkeyiq** [comment]
The diff for https://github.com/filesender/filesender/pull/1282/files is a little nasty to read for allowedPages(). Basically the new method starts with allow_pages_core and adds more allowed pages for guests, users, and admin. 

The plan is to only change things over what was there by allowing the privacy page to be viewed by default. Though the settings being taken from config should allow an admin to modify things a little more.

**Comment by monkeyiq** [comment]
I am happy to update things if there are other possibilities I might have missed here.

**Comment by monkeyiq** [comment]
I tried to use the constants instead of the strings directly in the update. I have been using the constants in ConfigDefaults.php too.

**Comment by monkeyiq** [comment]
Thank you for the detailed issue report :)

**Comment by peter-** [comment]
Thanks, the way you've implemented this looks like a useful generalisation and I see that `GUIPages::PRIVACY` is now allowed by default. I haven't tested this yet but lgtm.

Closing since this seems to be included in a release (v2.32) already.
